aptos-move/framework/move-stdlib/tests/mem_tests.move

```diff
@@ -0,0 +1,65 @@
+#[test_only]
+module std::mem_tests {
```

Comment:
The tests here are inadequate considering how important and dangerous the swap feature is.

We need to have full test coverage for all these swaps:
```
(container, container)

(indexed ref - generic, indexed ref - generic)
(indexed ref - specialized, indexed ref - specialized)
(indexed ref - generic, indexed ref - specialized)
(indexed ref - specialized, indexed ref - generic)
```
I would also encourage us add (rust-level) unit tests in `move-vm/types/src/values/value_tests`.

Reply:
I added some tests, but can add more. 

can you explain when does a &mut become each of the :

* indexer ref - generic
* indexed ref - specialized
* container